Packs/NetQuestOMX/Integrations/NetQuestOMX/NetQuestOMX.py

```diff
@@ -0,0 +1,327 @@
+
+import urllib3
+
+# Disable insecure warnings
+TOKEN_TTL = 7200  # in seconds (120 minutes token's expiration time)
+urllib3.disable_warnings()
+
+''' CONSTANTS '''
+
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'  # ISO8601 format with UTC, default in XSOAR
+DATE_FORMAT_FOR_TOKEN = "%m/%d/%Y, %H:%M:%S"
+VENDOR = "NetQuest"
+PRODUCT = "OMX"
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+
+    def __init__(self, base_url: str, credentials: dict, verify: bool, proxy: bool):
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy)
+        self.credentials = {"UserName": credentials["identifier"], "Password": credentials["password"]}
+        self.headers = {"Accept": "application/json"}
+        self.login()
+
+    def login(self):
+        """
+        In this method, the validity of the Access Token is checked, since the Access Token has a 120 minutes validity period.
+        Refreshes the token as needed.
+        """
+        now = datetime.utcnow()
+
+        if (cache := get_integration_context()) and cache.get("Token"):
+            expiration_time = datetime.strptime(
+                cache["expiration_time"], DATE_FORMAT_FOR_TOKEN
+            )
+
+            # check if token is still valid, continue using it. otherwise regenerate a new one
+            if (seconds_left := (expiration_time - now).total_seconds()) > 0:
+                demisto.debug(f"No need to regenerate the token, it is still valid for {seconds_left} more seconds")
+                return
+
+        demisto.debug("IntegrationContext token cache is empty or token has expired, regenerating a new token")
+        raw_response = self._refresh_access_token()
+        self._set_headers(raw_response)
+
+        set_integration_context(
+            {
+                "Token": self.headers["X-Auth-Token"],
+                "expiration_time": (
+                    now + timedelta(seconds=(TOKEN_TTL - 60))  # decreasing 60s from token expiry for safety
+                ).strftime(DATE_FORMAT_FOR_TOKEN),
+            }
+        )
+
+    def _refresh_access_token(self) -> dict:
+        """
+        Since the validity of the Access Token is 120 minutes, this method refreshes it and returns the new token json.
+
+        """
+
+        try:
+            response_json = self._http_request(
+                method="POST", url_suffix="/api/SessionService/Sessions", headers=self.headers, data=self.credentials
+            )
+        except Exception as e:
+            raise DemistoException(
+                "An error was occurred when creating a new token. Please verify your credentials."
+            ) from e
+        return response_json
+
+    def _set_headers(self, raw_response: dict):
+        """
+        This method is called during the client's building or when a new token is generated since the old one has expired.
+        """
+        # TODO - Ensure the access
+        self._headers["X-Auth-Token"] = raw_response["Headers"]["X-Auth-Token"]
```

Comment:
since it's a one-liner, and is only used in one place, I think we can just copy it to `login()`